gax/src/main/java/com/google/api/gax/batching/v2/BatchingSettings.java

```diff
@@ -0,0 +1,156 @@
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.batching.v2;
+
+import com.google.api.core.BetaApi;
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Preconditions;
+import javax.annotation.Nullable;
+import org.threeten.bp.Duration;
+
+/**
+ * Represents the batching settings to use for an API method that is capable of batching.
+ *
+ * <p>By default the settings are configured to <b>not</b> use batching (i.e. the batch size
+ * threshold is 1). This is the safest default behavior, which has meaning in all possible
+ * scenarios. Users are expected to configure actual batching thresholds explicitly: the element
+ * count, the request bytes count and the delay.
+ *
+ * <p>Warning: With the wrong settings, it is possible to cause long periods of dead waiting time.
+ *
+ * <p>When batching is turned on for an API method, a call to that method will result in the request
+ * being queued up with other requests. When any of the set thresholds are reached, the queued up
+ * requests are packaged together in a batch and set to the service as a single RPC. When the
+ * response comes back, it is split apart into individual responses according to the individual
+ * input requests.
+ *
+ * <p>There are several supported thresholds:
+ *
+ * <ul>
+ *   <li><b>Delay Threshold</b>: Counting from the time that the first message is queued, once this
+ *       delay has passed, then send the batch. The default value is 1 millisecond.
+ *   <li><b>Message Count Threshold</b>: Once this many messages are queued, send all of the
+ *       messages in a single call, even if the delay threshold hasn't elapsed yet. The default
+ *       value is 1 message.
+ *   <li><b>Request Byte Threshold</b>: Once the number of bytes in the batched request reaches this
+ *       threshold, send all of the messages in a single call, even if neither the delay or message
```

Comment:
...even if neither the delay nor the message count thresholds have been exceeded yet. (editor comment: neither is followed by nor, either is followed by or)